### PR TITLE
Fix bugs

### PR DESCRIPTION
**Pull request by sarthak-marwaha**
- Pass correct context in FObjectArray adapt
- Call FObjectArray adapt from AbstractSectionedDetailView
- Add reserveLabelSupport to props

### CODE REVIEW

File: src/foam/u2/detail/SectionView.js [hunk 1]
```diff
@@ -171,7 +171,7 @@ foam.CLASS({
                       return shown ? self.PropertyBorder.create({
                         prop: p,
                         data$: self.data$,
-                        reserveLabelSpace: true
+                        reserveLabelSpace: p.reserveLabelSpace ?? true
```
Comment (kgrgreer) [inline src/foam/u2/detail/SectionView.js:174]: Why is this necessary to have the "?? true"?

Comment (sarthak-marwaha) [reply src/foam/u2/detail/SectionView.js:174]: Yes, the `reserveLabelSpace` property doesnt actually exist on Property, it is just something that might be in viewConfig so it would be empty most of the time hence the ?? true
